modules/cnn_3dobj/samples/sphereview_3dobj_demo.cpp

```diff
@@ -0,0 +1,124 @@
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of Willow Garage, Inc. nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+#include <opencv2/cnn_3dobj.hpp>
+#include <opencv2/viz/vizcore.hpp>
+#include <iostream>
+#include <string>
+#include <stdio.h>
+#include <stdlib.h>
+using namespace cv;
+using namespace std;
+using namespace cv::cnn_3dobj;
+int main(int argc, char *argv[]){
+	const String keys = "{help | | demo :$ ./sphereview_test -radius=350 -ite_depth=1 -plymodel=../ape.ply -imagedir=../data/images_ape/, then press 'q' to run the demo for images generation when you see the gray background and a coordinate.}"
+			     "{radius | 350 | Distanse from camera to object, used for adjust view for the reason that differet scale of .ply model.}"
+			     "{ite_depth | 1 | Iteration of sphere generation, we add points on the middle of lines of sphere and adjust the radius suit for the original radius.}"
+			     "{plymodel | | path of the '.ply' file for image rendering. }"
+			     "{imagedir | | path of the generated images. }";
+	cv::CommandLineParser parser(argc, argv, keys);
+	parser.about("Demo for Sphere View data generation");
+	if (parser.has("help"))
+	{
+		parser.printMessage();
+		return 0;
+	}
+	//float radius = atof(argv[1]);
+	float radius = parser.get<float>("radius");
+	//int ite_depth = argv[2][0] - '0';
+	int ite_depth = parser.get<int>("ite_depth");
+	//std::string plymodel = argv[3];
+	string plymodel = parser.get<string>("plymodel");
+	string imagedir = parser.get<string>("imagedir");
+	cv::cnn_3dobj::IcoSphere ViewSphere(10,ite_depth);
```

Comment:
After you create the IcoSphere object, you should check for pose duplicates before iterating over them to render the object. If you see the result images, you'll get a lot of duplicates.
You can easily avoid them using a Map or something like that